content/ui-patterns/overlays.mdx

```diff
@@ -0,0 +1,415 @@
+</Box>
+
+<LabelGroup display="block" mb={4}>
+    <Label as="a" href="https://primer.style/react/Overlay" variant="xl" color="text.success" outline style="text-decoration: none; line-height: 20px;">
+        <img width="20" height="20" alt=" " src="https://user-images.githubusercontent.com/293280/123878374-ce9d4d00-d8f3-11eb-8adf-1a160292ff53.png" style="vertical-align: middle; margin-right: 4px;" />
+        React
+    </Label>
+    <Label as="a" href="https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=9677%3A44413" variant="xl" outline style="text-decoration: none; line-height: 20px;">
+        <img width="20" height="20" alt=" " src="https://user-images.githubusercontent.com/293280/123878720-80d51480-d8f4-11eb-9b10-d02a1cb606f8.png" style="vertical-align: middle; margin-right: 4px;" />
+        Figma
+    </Label>
+</LabelGroup>
+
+<img src="https://user-images.githubusercontent.com/293280/125706076-b16bb3d4-800f-4310-a082-f576e215e6c8.png" alt="Overlay examples" />
+
+## Overview
+
+Overlays can streamline the interface by allowing extra options to be displayed as needed. Overlays create a new modality to the user, and can facilitate the completion of individual tasks.
+
+**Keep overlays simple**. Overlays are meant to be interacted with quickly. Avoid creating a whole page inside an overlay. Prefer single-column layouts that convey a clear goal the moment the overlay is opened.
+
+**Keep overlays accessible**. Overlays should be fully functional with keyboard. When an overlay is present, the rest of the page becomes noninteractive until the overlay is dismissed. Pressing Escape should dismiss the overlay. When an overlay is dismissed, focus generally returns to the element that opened it.
+
+**Overlays should work well on any device**. Overlays must adjust their designs to fit in smaller viewports, and may adapt their style to become bottom sheets or full-screen on narrow devices. Make sure the overlay contents work with all supported sizes and input methods.
+
+## Anatomy
+<ImageContainer mb={4}>
+  <img
+    width="636"
+    src="https://user-images.githubusercontent.com/293280/146623655-e8cda46c-10d5-4473-b54b-278bde283913.png"
+    alt="Overlays have 3 regions: header, content, and footer. These regions may be divided by a full-width divider. Overlays use shadows to indicate elevation, and may appear above a backdrop that dims the rest of the screen."
+    style="background: none"
+  />
+</ImageContainer>
+<Caption>Anatomy of an overlay.</Caption>
+
+**Keep the edges of an overlay free of text and heavy visuals.** This breathing room is meant to visually distinguish the overlay as an elevated surface, alongside its shadow. As a rule of thumb, a safe area of 16px is enough. Components that have transparent or discrete backgrounds may take into account their own paddings, and appear beyond the safe area.
```

Comment:
> Can the safe area margin be customized? If not, I question if this is the correct wording. If there are use cases when a designer would increase the safe area margin, can we show those or list them?

The overlay contents is a blank canvas, thinking about how it's implemented. The safe area is a recommendation, as it really depends on how the contents inside are placed.

I reviewed the sentence, went from this:
```
As a rule of thumb, a safe area of 16px is enough.
```
to this
```
Generally, a compact a safe area of 16px is preferred.
```

What do you think? 🙇 

> An example image for this would be good to see.

I've explored this, but it doesn't seem to belong in this first part of the Anatomy section:

![Screen Shot 2022-01-14 at 16 46 41](https://user-images.githubusercontent.com/293280/149602099-11a3dbc8-8458-499b-b5fd-20975fcb5ded.png)

<details>

```mdx
<ImageContainer mb={4}>
  <img
    width="636"
    src="https://user-images.githubusercontent.com/293280/149601850-2736ea96-b6ab-46b7-8bdb-11920f833d29.png"
    alt=""
    style="background: none"
  />
</ImageContainer>
<Caption>

An example of a dialog with 16px paddings. Notice the **Close** button is closer to the edges, but its icon is still within the safe area.

</Caption>
```

</details>